sui/typescript/dapp_kit_example/src/client/App.tsx

```diff
@@ -0,0 +1,175 @@
+      }  
+  } catch (e) {
+    console.log(e);
+  }
+  }
+
+
+  // Poll the Full node represented by the SuiClient until the given digest
+  // has been checkpointed and propagated to the node, and the node returns 
+  // results for the digest. On the response, update the page accordingly.
+  const waitForTxAndUpdateResult = async (digest: string) => {
+    const finalResult = await shinamiClient.waitForTransaction({ 
+      digest: digest,
+      options: {
+        showEffects: true,
+        showEvents: true
+      }
+    });
+
+    if (finalResult.effects && finalResult.events && finalResult.effects.status.status == "success"){
+      const resultObj = finalResult.events[0].parsedJson as AddCallEvent;
+      const result = resultObj.result;
+      setLatestDigest(digest);
+      setLatestResult(result);
+      setnewSuccessfulResult(true);
+    } else {
+      console.log("Transaction did not execute successfully.");
+    }
+  }
+
+
+  // 1. Ask the backend to build and sponsor a Move call transction with the given user input.
+  // 2. Sign the sponsored transaction returned from the backend with the user's connected wallet.
+  // 3. Ask the backend to execute the signed transaction.
+  // 4. Return the SuiTransactionBlockResponse to the caller.
+  const connectedWalletTx = async (x: number, y: number, senderAddress: string): Promise<SuiTransactionBlockResponse> => {
+    const sponsorshipResp = await axios.post('/buildSponsoredtx', {
+      x: x,
+      y: y,
+      sender: senderAddress
+    });
+
+    const { signature } = await signTransaction({
+      transaction: sponsorshipResp.data.txBytes
+    });
+
+    const resp =  await axios.post('/executeSponsoredTx', {
+      tx: sponsorshipResp.data.txBytes,
+      sponsorSig: sponsorshipResp.data.signature,
+      senderSig: signature
+    });
+    return resp.data;
+  }
+
+
+  // 1. Ask the backend to build, sponsor, sign, and execute a Move call transaction with the 
+  //    given user input. The sender is the user's Invisible Wallet, which in this example app
+  //    is just a hard-coded wallet for simplicity.
+  // 2. Return the SuiTransactionBlockResponse to the caller.
+  const invisibleWalletTx = async (x: number, y: number): Promise<SuiTransactionBlockResponse> => {
+    const resp = await axios.post('/invisibleWalletTx', {
+      x: x,
+      y: y,
+      userId: "abc123"
+    });
+    return resp.data;
+}
+
+
+
+  return (
+    <>
+      <Flex
+        position="sticky"
+        px="4"
+        py="2"
+        justify="between"
+      >
+        <Box>
+          <Heading>Shinami Sponsored Transactions with dApp Kit</Heading>
+        </Box>
+        <Box>
+          <h3>Pick two integers to add in a Move call</h3>
+          <form onSubmit={executeTransaction}>
+            <div>
+              <label htmlFor="integerOne">First integer:</label>
+              <input type="text" name="integerOne" id="integerOne" required />
```

Comment:
Change type to "number". This will prevent users from entering text causing the app to throw an error

Reply:
updated, thanks